libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1710,19 +1710,23 @@ Type const* TypeChecker::typeCheckTypeConversionAndRetrieveReturnType(
 		{
 			if (auto argArrayType = dynamic_cast<ArrayType const*>(argType))
 			{
-				auto resultArrayType = dynamic_cast<ArrayType const*>(resultType);
-				solAssert(!!resultArrayType, "");
-				solAssert(
-					argArrayType->location() != DataLocation::Storage ||
-					(
+				if (auto resultArrayType = dynamic_cast<ArrayType const*>(resultType))
```

Comment:
Holy... this assertion is complicated.

Reply:
I didn't change it, but yes, it is too complicated. I can maybe refactor it?

Reply:
Pulling the variable declaration inside does not make it simpler :)

Reply:
Ah sorry, I thought it was an `if (x; y)`